### DISCUSSION TITLE
NX and Markdoc

### DISCUSSION THREAD

**Discussion by dbani-dev** [markdoc/markdoc#58] [Q&A] [answered]
Has anyone setup [NX](https://nx.dev/packages/next#setting-up-nextjs) + Markdoc before? Running `nx serve docs` works and runs app with `next.config.js` configuration:

```
const withMarkdoc = require('@markdoc/next.js');
module.exports = withMarkdoc()({
  pageExtensions: ['js', 'md'],
});
```

However it seems custom markdoc `tags`, `nodes`  etc don’t seem to be working while running with NX?

**Comment by dbani-dev** [comment, marked as answer]
Nevermind, it seems the `schemaPath` needed to be set :thumbsup:

**Reply by mfix-stripe** [reply]
@davidbani can you share your implementation here so future readers can see the solution? That would be much appreciated! 🙏

**Reply by dbani-dev** [reply]
@mfix-stripe sure. In order to run NX + markdoc, the package app requires the `schemaPath` to be set in the `next.config.js`:

```js
module.exports = withMarkdoc({
  schemaPath: './packages/docs/markdoc',
})({
  pageExtensions: ['js', 'jsx', 'ts', 'tsx', 'md', 'mdoc'],
});
```
Running the NX package app should work as expected.
